src/caches.jl

```diff
@@ -131,20 +140,37 @@ function OrdinaryDiffEq.alg_cache(
 
     # Measurement model related things
     R = zeros(uElType, d, d)
-    H = zeros(uElType, d, D)
+    H = if KRONECKER
+        copy(E1)
+    else
+        zeros(uElType, d, D)
+    end
     v = zeros(uElType, d)
-    S = PSDMatrix(zeros(uElType, D, d))
+    S = if KRONECKER
+        PSDMatrix(kronecker(_I(d), zeros(uElType, q + 1)))
+    else
+        PSDMatrix(zeros(uElType, D, d))
+    end
     measurement = Gaussian(v, S)
 
     # Caches
     du = f isa DynamicalODEFunction ? similar(u[2, :]) : similar(u)
     ddu = zeros(uElType, length(u), length(u))
     pu_tmp =
         f isa DynamicalODEFunction ?
-        Gaussian(zeros(uElType, 2d), PSDMatrix(zeros(uElType, D, 2d))) : copy(measurement)
+        Gaussian(zeros(uElType, 2d), PSDMatrix(
```

Comment:
Refactor this to be more readable. Maybe even just remove the `copy(measurement)` shortcut?